platform/barefoot/sonic-platform-modules-imt/sonic_platform/chassis.py

```diff
@@ -0,0 +1,303 @@
+        return self.__fan_drawers
+
+    @_fan_drawer_list.setter
+    def _fan_drawer_list(self, value):
+        pass
+
+    @property
+    def _thermal_list(self):
+        if self.__thermals is None:
+            self.__thermals = thermal_list_get()
+        return self.__thermals
+
+    @_thermal_list.setter
+    def _thermal_list(self, value):
+        pass
+
+    @property
+    def _psu_list(self):
+        if self.__psu_list is None:
+            self.__psu_list = psu_list_get()
+        return self.__psu_list
+
+    @_psu_list.setter
+    def _psu_list(self, value):
+        pass
+
+    @property
+    def _sfp_list(self):
+        if self.__sfp_list is None:
+            self.__sfp_list = sfp_list_get()
+        return self.__sfp_list
+
+    @_sfp_list.setter
+    def _sfp_list(self, value):
+        pass
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the chassis
+        Returns:
+            bool: True if chassis is present, False if not
+        """
+        return True
+
+    def get_sfp(self, index):
+        """
+        Retrieves sfp represented by (1-based) index <index>
+
+        Args:
+            index: An integer, the index (1-based) of the sfp to retrieve.
+                   The index should be the sequence of a physical port in a chassis,
+                   starting from 1.
+                   For example, 0 for Ethernet0, 1 for Ethernet4 and so on.
+
+        Returns:
+            An object dervied from SfpBase representing the specified sfp
+        """
+        sfp = None
+
+        try:
+            sfp = self._sfp_list[index-1]
+        except IndexError:
+            sys.stderr.write("SFP index {} out of range (1-{})\n".format(
+                             index, len(self._sfp_list)-1))
+        return sfp
+
+    def get_status(self):
+        """
+        Retrieves the operational status of the chassis
+        Returns:
+            bool: A boolean value, True if chassis is operating properly
+            False if not
+        """
+        return True
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot
+
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must be
+            one of the predefined strings in this class. If the first string
+            is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be used
+            to pass a description of the reboot cause.
+        """
+        return self.REBOOT_CAUSE_NON_HARDWARE, ''
+
+    ##############################################
+    # Fan methods
+    ##############################################
+
+    def get_num_fans(self):
+        """
+        Retrieves the number of fans available on this chassis
+
+        Returns:
+            An integer, the number of fan modules available on this chassis
+        """
+        return len(self._fan_list)
```

Comment:
I'm not sure how we work with _fan_list. E.g. for __fan_drawers attribute of Chassis class we have getter property _fan_drawer_list. This getter return list of FanDrawer objects constructed by fan_drawer_list_get public function. But we never did the same for _fan_list. Basically each object of FanDrawer has it's own _fan_list and doesn't look that we reach it here.

Reply:
Yeah, you're right. fixed.